website/addons/s3/static/s3NodeConfig.js

```diff
@@ -91,31 +91,45 @@ ViewModel.prototype.updateBucketList = function(){
 
 ViewModel.prototype.selectBucket = function() {
     var self = this;
+
+    // The current version of boto does not handle bucket names containing periods
+    var isValidBucket = /^(?!.*(\.\.|-\.))[^.][a-z0-9\d-]{2,61}[^.]$/;
```

Comment:
Hm, then maybe add a parameter to the `isValidBucket` function so you can do `isValidBucket(bucketName, {allowPeriods: true})`?